core/Requirement/Prohibited_categories.php

```diff
@@ -0,0 +1,192 @@
+<?php
+/**
+ * @package     PublishPress\Checklists
+ * @author      PublishPress <help@publishpress.com>
+ * @copyright   copyright (C) 2019 PublishPress. All rights reserved.
+ * @license     GPLv2 or later
+ * @since       1.0.0
+ */
+
+namespace PublishPress\Checklists\Core\Requirement;
+
+use PPCH_Checklists;
+
+defined('ABSPATH') or die('No direct script access allowed.');
+
+class Prohibited_categories extends Base_multiple
+{
+    /**
+     * The name of the requirement, in a slug format
+     *
+     * @var string
+     */
+    public $name = 'prohibited_categories';
+
+    /**
+     * @var int
+     */
+    public $position = 50;
+
+    /**
+     * @var string
+     */
+    private $DELIMITER = '__';
+
+    /**
+     * Initialize the language strings for the instance
+     *
+     * @return void
+     */
+    public function init_language()
+    {
+        $this->lang['label']          = __('Prohibited categories: %s', 'publishpress-checklists');
+        $this->lang['label_settings'] = __('Prohibited categories', 'publishpress-checklists');
+    }
+
+    /**
+     * Returns the current status of the requirement.
+     *
+     * @param stdClass $post
+     * @param mixed $option_value
+     *
+     * @return mixed
+     */
+    public function get_current_status($post, $option_value)
+    {
+        $categories = wp_get_post_categories($post->ID);
+        $option_ids = $this->category_parser($option_value, 0);
+
+        return empty(array_intersect($option_ids, $categories));
+    }
+
+    /**
+     * Get Categories Hierarchical List
+     * 
+     * @param array $args
+     * @return WP_Term[]
+     */
+    private function _get_categories_hierarchical($args = array())
+    {
+        if( !isset( $args[ 'parent' ] ) ) $args[ 'parent' ] = 0;
+
+        $categories = get_categories( $args );
+        foreach( $categories as $key => $category ) {
+            $args['parent'] = $category->term_id;
+            $categories[$key]->children = $this->_get_categories_hierarchical($args);
+        }
+
+        return $categories;
+    }
+
+    /**
+     * Transform categories to labels
+     * 
+     * @param WP_Term[] $categories
+     * @return String[] $labels
+     */
+    private function _transform_categories($categories = array())
```

Comment:
Hi @richaferry I understand the purpose of underscore is to signify it's a private class but I think you should follow current plugin pattern and coding convention.

Reply:
Hi @olatechpro,
Sorry for late reply. 
Thankyou for your advice for this. Im already changes the naming convention for method. Please kind help to recheck my pr. Need your advice.

Reply:
Im also already provide test link. If you want to recheck about this changes. link test [here](https://app.instawp.io/wordpress-auto-login?site=$2y$10$mxvVSXWSjatBapSppd2e2uKO.XAgkJ3F..UH/ckxBTsHH613ZQfDK)